### PR TITLE
Round

### PR DESCRIPTION
**Pull request by jtlap**
I found a error due to the fact Half+prev(Half) is not strictly less than one for float and double!

### CODE REVIEW

File: test/function/simd/round.cpp [hunk 1]
```diff
@@ -30,7 +34,8 @@ void test(Env& $)
   }
   p_t aa1(&a1[0], &a1[0]+N);
   p_t bb (&b[0], &b[0]+N);
-  STF_IEEE_EQUAL(bs::round(aa1), bb);
+  std::cout << aa1 << std::endl;
```
Comment (jfalcou) [inline test/function/simd/round.cpp]: useless display

File: include/boost/simd/arch/common/simd/function/round.hpp [hunk 2]
```diff
@@ -57,10 +59,9 @@ namespace boost { namespace simd { namespace ext
    {
       BOOST_FORCEINLINE A0 operator()( const A0& a0) const BOOST_NOEXCEPT
       {
-        auto isf = logical_andnot(is_flint(a0+a0), is_flint(a0));
-        auto r1 = round2even(a0);
-//        if (none(isf)) return r1;
-        return if_else(isf, -trunc(-a0), r1);
+        const A0 v = bs::abs(a0);
+        const A0 c = bs::ceil(v);
+        return if_else(v > Maxflint<A0>(), a0, copysign(if_dec(c-Half<A0>() > v, c), a0));
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/round.hpp:64]: what's the performance impact ?

Comment (jtlap) [inline include/boost/simd/arch/common/simd/function/round.hpp:64]: new
         1         14.3081      (cpe)   (median)           14146      rand<pack<float, 4>>[-10, 10]
         1         18.6676      (cpe)   (median)           26816     rand<pack<double, 2>>[-10, 10]
old
         1         8.95688      (cpe)   (median)           15838   rand<pack<float, 4>>[-10, 10]
         1         17.6731      (cpe)   (median)           27491   rand<pack<double, 2>>[-10, 10]

but the new is correct the old is not.
However nearbyint is quite 10 times faster

Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/round.hpp:64]: that's for scalar or SIMD ?

**Pull request merged**